interface/product_registration/product_registration_service.php

```diff
@@ -0,0 +1,87 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
+include_once("../globals.php");
+require_once($GLOBALS['fileroot'] . "/interface/main/exceptions/invalid_email_exception.php");
+require_once($GLOBALS['fileroot'] . "/interface/product_registration/local_product_registration_exception.php");
+require_once($GLOBALS['fileroot'] . "/interface/product_registration/remote_product_registration_exception.php");
+require_once($GLOBALS['fileroot'] . "/ccr/uuid.php");
+
+// TODO: @Brady to help me get this in the migration DSL...
+// TODO: Figure out why foreign key returns `errno 150`... am thinking it's because user email isn't uniq.
+//CREATE TABLE product_registration (
+//    registration_id CHAR(36) NULL,
+//    email VARCHAR(255) NULL,
+//    opt_out TINYINT(1) NULL,
+//    PRIMARY KEY (registration_id)/*
+//    FOREIGN KEY (email) REFERENCES users(email)*/
+//);
+
+class ProductRegistrationService {
+    public function __construct() {}
+
+    public function getProductStatus() {
+        $row = sqlQuery("SELECT * FROM product_registration LIMIT 1;");
+
+        if (empty($row)) {
+            return 'UNREGISTERED';
+        } else if (!(empty($row['registration_id']))) {
+            return 'REGISTERED';
+        } else if (!(empty($row['opt_out'])) && $row['opt_out'] == true) {
+            return 'OPT_OUT';
+        }
+    }
+
+    public function registerProduct($email) {
+        if (empty($email)) {
+            $this->optOutStrategy();
+            return null;
+        } else {
+            return $this->optInStrategy($email);
+        }
+    }
+
+    private function optInStrategy($email) {
+        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+            throw new InvalidEmailException($email);
+        }
+
+        $uuid = getUuid();
+
+        // TODO: Is there support for parameterization for security purposes?
+        $result = sqlStatement("INSERT INTO product_registration VALUES ('" . $uuid . "','" . $email . "', false);");
+
+        // TODO: how can I check the outcome of the response?
+        if (empty($result)) {
+            throw new LocalProductRegistrationException("Database issue... can't register product.");
+        }
+
+        // @Robert or @Visolve: remote server call here... RemoteProductRegistrationException("Can't register product with server.")
+
+        return $uuid;
+    }
+
+    // void... don't bother checking for success/failure.
+    private function optOutStrategy() {
+        sqlStatement("INSERT INTO product_registration VALUES ('', null, true);");
```

Comment:
since you never know where the table is going to head, should explicitly list the sql-columns that are going to be populate; INTO product_registration (...) VALUES (...)